client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+export const requestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( options => dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+		options,
```

Comment:
I'm so happy to find this code so well written that I can focus on higher-level requests :smile:

At this point we're in a juncture: we're setting between the Calypso world and the API world and this `options` sticks out like a sore thumb.

What kind of data is it?
What is its shape?

One piece that I think is critical in the API middleware is a pair of functions `toApi()` and `fromApi()` which perform the data translation to bridge the idiomatic gap. In this case since we're not issuing data to the API we don't have any need for `toApi()`.

Here's an example that could be right based on the tests in this PR:

```js
const fromApi = data => ( {
	primaryEmail: data.primary_email,
	primarySms: data.primary_sms,
	secondaryEmail: data.secondary_email,
	secondarySms: data.secondary_sms,
} );
```

but here's a real kicker. there's no reason we have to make the data in Calypso look like the data from the API. I don't know how we want to use this in calypso, but maybe we store the contact info as a prioritized list by list position:

```js
const fromApi = data => [ {
	email: data.primary_email,
	sms: data.primary_sms,
}, {
	email: data.secondary_email,
	sms: data.secondary_sms,
} ];
```

we can shape the data however we want for it to be used and not simply stick with the shape of how it was given to us. I would encourage you to consider how this is supposed to be used and in which format it's most appropriate then map it at this interface between the API and Calypso.

this also will make it easier to mock data and swap it out. if we use the console dispatcher, for instance, do we want to have to look up the API code in PHP to figure out what kinds of data and names it returns or should we instead have a good place declared in JS that shows what kind of data structure we need to end up with and just recreate it?